web/styles/right_sidebar.css

```diff
@@ -363,38 +363,55 @@ $user_status_emoji_width: 24px;
     grid-template-rows: var(--line-height-sidebar-row-prominent);
     grid-template-columns: minmax(0, 1fr) auto;
     align-items: center;
-    margin-right: var(--width-simplebar-scroll-hover);
+    margin-bottom: 10px;
 
     #userlist-header-search {
         display: grid;
         grid-template-rows: var(--line-height-sidebar-row-prominent);
-        grid-template-columns: minmax(0, 1fr) 20px;
+        grid-template-columns: minmax(0, 1fr) 30px;
         align-items: center;
-    }
 
-    #userlist-title {
-        margin: 0;
-    }
+        & .user-list-filter {
+            grid-area: 1 / 1 / 2 / 3;
+            white-space: nowrap;
+            text-overflow: ellipsis;
+            overflow: hidden;
+            /* Prevent text from colliding with #clear_search_button */
+            padding-right: 28px;
+            /* Push back against inherited styles; let CSS Grid be in
+               charge of the height. */
+            height: auto;
+        }
 
-    #user_filter_icon {
-        opacity: 0.5;
-        justify-self: center;
+        #clear_search_people_button {
+            grid-area: 1 / 2 / 2 / 3;
```

Comment:
Yeah, I think we should have a named grid area or at least line names. I'm not sure. But I don't like having to reason about this in my head.

Reply:
(Also it occurs to me that at least some of this might be stuff that I wrote originally 🙃).

Reply:
Yeah I just copied this all from below hehe. Would you be down to help me refactor it? I'm not super familiar with grid CSS best practices.

Reply:
I'm making a note to work on this probably as part of the new user-circles PR. But thank you!